src/renderer/tr_shader.cpp

```diff
@@ -232,10 +237,15 @@ void R_RemapShader(const char *shaderName, const char *newShaderName, const char
 	hash = generateHashValue(strippedName, FILE_HASH_SIZE);
 	for (sh = hashTable[hash]; sh; sh = sh->next) {
 		if (Q_stricmp(sh->name, strippedName) == 0) {
-			if (sh != sh2) {
+			if (sh->remappedShader && sh->advancedRemap) {
+				// Advanced remaps take priority over the old ones
+				continue;
+			} else if (sh != sh2) {
```

Comment:
Have you tried making a shader remapping cycle? sh1 -> sh2 -> sh1 Also what if sh1 has advanced remap to sh2 and sh2 has basic remap to sh3? Maybe it would be safer to just use either advanced or basic, but not mix them together?

Reply:
That is not an issue, because `RB_BeginSurface` only checks for a single remap, no recursion: https://github.com/mvdevs/jk2mv/blob/da97452e607bad19a13da4660d6442f7dc3a8dc6/src/renderer/tr_shade.cpp#L324

Due to this the remap `sh1 -> sh2 -> sh1` effectively makes all original `sh1` surfaces show up as `sh2` and vice versa. This enables you to swap shaders. Vanilla remaps already worked this way.

I originally set the `advancedRemap` value on the destination shader, but that would have been problematic, because different original shaders could point to the same remap. A shader itself on the other hand can only ever point to a single remap, that's why I decided to set it on the original shader. This way mixing advanced remaps with classic ones is not a problem.